app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt

```diff
@@ -132,9 +134,22 @@ fun HikeDetailScreen(
 
   val selectedHike by hikesViewModel.selectedHike.collectAsState()
 
+  // Gets initialized here so that the LaunchedEffect has access to it. The value is only actually
+  // initialised and used in HikeDetailsContent
+  val mapViewState = remember { mutableStateOf<MapView?>(null) }
+
+  // If the selected hike is null, save the map's state and go back to the map screen
```

Comment:
Suggestion: You can add a Launched effect directly in `HikeDetailsContent`

Reply:
I am not sure that that will work. We could pass `selectedHike` as a parameter to the `HikeDetailsContent`, and have the `LaunchedEffect` there, but then all the code that gets executed before the `LaunchedEffect`, especially this line: `val detailedHike = selectedHike!!.withDetailsOrThrow()` will execute when selectedHike may be null.

Reply:
See my comment in the above conversation.